llvm/lib/Analysis/ValueTracking.cpp

```diff
@@ -793,7 +801,7 @@ static void computeKnownBitsFromCond(const Value *V, Value *Cond,
                                      KnownBits &Known, unsigned Depth,
                                      const SimplifyQuery &SQ, bool Invert) {
   Value *A, *B;
-  if (Depth < MaxAnalysisRecursionDepth &&
+  if (Depth < getAnalysisRecursionDepthLimit() &&
```

Comment:
Avoid calling this helper function multiple times inside a function. Compute it once and then reuse the result. This PR contains many such instances.

Reply:
Oops -- still need to address this.